internal/report/json/json.go

```diff
@@ -0,0 +1,66 @@
+package json
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/rs/zerolog/log"
+	"github.com/saucelabs/saucectl/internal/msg"
+	"github.com/saucelabs/saucectl/internal/report"
+)
+
+// Reporter represents struct to report in json format
+type Reporter struct {
+	WebhookURL string
+	Filename   string
+	Results    []report.TestResult
+}
+
+// Add adds a TestResult
+func (r *Reporter) Add(t report.TestResult) {
+	r.Results = append(r.Results, t)
+}
+
+// Render sends the result to specified webhook WebhookURL and log the result to the specified json file
+func (r *Reporter) Render() {
+	body, err := json.Marshal(r.Results)
+	if err != nil {
+		log.Error().Msgf("failed to generate test result (%v)", err)
+		return
+	}
+
+	if r.WebhookURL != "" {
+		resp, err := http.Post(r.WebhookURL, "application/json", bytes.NewBuffer(body))
+		if err != nil {
+			log.Error().Msgf("failed to send result (%v)", err)
+		} else {
+			if resp.StatusCode >= http.StatusInternalServerError {
+				log.Error().Msg(msg.InternalServerError)
+			}
```

Comment:
This part is still very technical. Unlike https://github.com/saucelabs/saucectl/pull/529/files#diff-ecd8d3b8194f17a2e35d29de9a04a6bc1cf858247148383af3839e6296c77851R45